### ISSUE TITLE
#1580 causes type issue in `MaybeOptionalInit`

### ISSUE THREAD

**Issue by mikestopcontinues** [drwpow/openapi-typescript#1584]
`MaybeOptionalInit` [now requires](https://github.com/drwpow/openapi-typescript/blame/b0d2bb2da82a30d4b3c04968c3fb6296f873fd4e/packages/openapi-fetch/src/index.d.ts#L174) Paths[P] to be a _complete_ record including every `HttpMethod`. However, routes rarely have every method. I think it should be defined as `Partial<Record<HttpMethod, {}>>` instead. The same with `ClientMethod`:

```ts
export type MaybeOptionalInit<
  P extends Partial<Record<HttpMethod, {}>>,
  M extends keyof P,
> =
  HasRequiredKeys<FetchOptions<FilterKeys<P, M>>> extends never
    ? [(FetchOptions<FilterKeys<P, M>> | undefined)?]
    : [FetchOptions<FilterKeys<P, M>>];

export type ClientMethod<
  Paths extends Record<string, Partial<Record<HttpMethod, {}>>>,
  M extends HttpMethod,
> = <
  P extends PathsWithMethod<Paths, M>,
  I extends MaybeOptionalInit<Paths[P], M>,
>(
  url: P,
  ...init: I
) => Promise<FetchResponse<Paths[P][M], I[0]>>;

```

**Comment by drwpow** [comment]
Ah yes you’re right! Sad that the internal type tests didn’t catch this. I’d accept a PR for this if you (or anyone else) is willing to add it
